open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/enterprise/repositoryconnector/executors/ClassifyEntityProxyExecutor.java

```diff
@@ -0,0 +1,164 @@
+/* SPDX-License-Identifier: Apache 2.0 */
+/* Copyright Contributors to the ODPi Egeria project. */
+package org.odpi.openmetadata.repositoryservices.enterprise.repositoryconnector.executors;
+
+
+import org.odpi.openmetadata.frameworks.auditlog.AuditLog;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.OMRSMetadataCollection;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.instances.EntityProxy;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.instances.InstanceProperties;
+import org.odpi.openmetadata.repositoryservices.enterprise.repositoryconnector.accumulators.MaintenanceAccumulator;
+import org.odpi.openmetadata.repositoryservices.ffdc.exception.ClassificationErrorException;
+import org.odpi.openmetadata.repositoryservices.ffdc.exception.EntityNotKnownException;
+import org.odpi.openmetadata.repositoryservices.ffdc.exception.FunctionNotSupportedException;
+import org.odpi.openmetadata.repositoryservices.ffdc.exception.InvalidParameterException;
+import org.odpi.openmetadata.repositoryservices.ffdc.exception.PropertyErrorException;
+import org.odpi.openmetadata.repositoryservices.ffdc.exception.RepositoryErrorException;
+import org.odpi.openmetadata.repositoryservices.ffdc.exception.UserNotAuthorizedException;
+
+/**
+ * ClassifyEntityProxyExecutor provides the executor for the classifyEntityProxy method.
+ * This is a tricky request because the entity has been deleted so it is
+ * not retrievable until restored.  Also, this is an optional function
+ * so the repository where the entity has its home may not support restore.
+ *
+ * The only possible approach is to step through the repositories hoping that one
+ * will respond positively.
+ */
+public class ClassifyEntityProxyExecutor extends RepositoryExecutorBase {
+
+    private final MaintenanceAccumulator accumulator;
+    private final String                 entityProxyGUID;
+    private final String                 classificationName;
+    private final InstanceProperties     classificationProperties;
+
+    private EntityProxy                  updatedEntityProxy;
+
+    /**
+     * Constructor takes the parameters for the request.
+     *
+     * @param userId unique identifier for requesting user
+     * @param entityProxyGUID unique identifier (guid) for the entity
+     * @param classificationName String name for the classification
+     * @param classificationProperties list of properties to set in the classification
+     * @param auditLog logging destination
+     * @param methodName calling method
+     */
+    public ClassifyEntityProxyExecutor(String               userId,
+                                       String               entityProxyGUID,
```

Comment:
Rather than have new execurotr - could the old executors be updated with new method?